File: freight-triage-polyglot-ledger-recovery/environment/native/src/norm/reverse_bytes.cpp
#include "freight/normalize.h"

#include <algorithm>

namespace freight {

// Normalizer: reverse bytes.
std::string norm_reverse_bytes(const std::string& text) {
  std::string out(text);
  std::reverse(out.begin(), out.end());
  return out;
}

}  // namespace freight

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/ReverseBytes.java
package com.freight.norm;

/** Normalizer: reverse bytes. */
public final class ReverseBytes implements Normalizer {

    @Override
    public String name() {
        return "reverse_bytes";
    }

    @Override
    public String apply(String text) {
        return new StringBuilder(text).reverse().toString();
    }
}
